### ISSUE TITLE
Inlay hints don't support rest parameters for tuple types

### ISSUE THREAD

**Issue by looeee** [microsoft/TypeScript#45868]
Issue Type: <b>Bug</b>

Inlay hints don't support rest params. 
For example, I'm using [this method](https://github.com/yomotsu/camera-controls#setposition-positionx-positiony-positionz-enabletransition-) which has the following signature:

``` ts
controls.setPosition(x: number, y: number, z: number, enableTransition: boolean);
```

And I want to pass in a `Vector3Tuple` as the position, say `[0,0,0]`:

``` ts
type Vector3Tuple = [number, number, number];

const position: Vector3Tuple = [0,0,0];

controls.setPosition(...position, false);
```

This is what I see for the inlay hints:

![inlay rest](https://user-images.githubusercontent.com/5307958/133195439-d297413e-4705-4fea-9b84-b7d2b8c7a662.png)

In other words, `...position` is showing as a single argument when it should be three. 

If I expand the arguments, it does correctly show the hints:

![inlay explicit](https://user-images.githubusercontent.com/5307958/133195712-449eae12-4b00-43b8-b239-dcb678814fcf.png)

I'm not sure what the inlay hints should show for `...position`, however, the final parameter of `false` should correctly show `enableTransition`. 

VS Code version: Code 1.60.0 (e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff, 2021-09-01T10:41:52.311Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No
Remote OS version: Linux x64 5.10.16.3-microsoft-standard-WSL2

**Comment by sandersn** [comment]
I looked at the original inlay hints PR; there are only tests for rest parameters, but no spread arguments. The right design isn't obvious to me; we need to think about how it should work.
